unity-renderer/Assets/DCLPlugins/OutlinerPlugin/OutlinerController.cs

```diff
@@ -0,0 +1,142 @@
+
+    public OutlinerController(DataStore_Outliner dataStore, OutlineRenderersSO outlineRenderersSO)
+    {
+        this.dataStore = dataStore;
+        this.outlineRenderersSO = outlineRenderersSO;
+
+        (outlineMaskFeature, outlineScreenEffectFeature) = GetOutlineRenderFeatures();
+        outlineMaskFeature.SetActive(true);
+        outlineScreenEffectFeature.SetActive(true);
+        outlineScreenEffectFeature.settings.effectFade = 0;
+        shouldShow = false;
+
+        this.dataStore.avatarOutlined.OnChange += OnAvatarOutlinedChange;
+    }
+
+    private (OutlineMaskFeature maskFeature, OutlineScreenEffectFeature screenEffectFeature) GetOutlineRenderFeatures()
+    {
+        // Unity refuses to make this accessible.
+        // Some alternatives could be moving ForwardRenderer to Resources
+        // Have an ScriptableObject with all the references somewhere.
+        // I find reflection cleaner.
+
+        var pipeline = ((UniversalRenderPipelineAsset)GraphicsSettings.renderPipelineAsset);
+        FieldInfo propertyInfo = pipeline.GetType().GetField("m_RendererDataList", BindingFlags.Instance | BindingFlags.NonPublic);
+        var scriptableRendererData = ((ScriptableRendererData[])propertyInfo?.GetValue(pipeline))?[0];
+
+        var maskFeature = scriptableRendererData?.rendererFeatures.OfType<OutlineMaskFeature>().First();
+        var screenFeature = scriptableRendererData?.rendererFeatures.OfType<OutlineScreenEffectFeature>().First();
+
+        return (maskFeature, screenFeature);
+    }
+
+    private void OnAvatarOutlinedChange((Renderer renderer, int meshCount, float avatarHeight) current, (Renderer renderer, int meshCount, float avatarHeight) previous)
+    {
+        if (current.renderer != null)
+        {
+            outlineRenderersSO.avatar = current;
+            //We are currently showing (or animating towards showing), nothing else to do
+            if (shouldShow)
+                return;
+
+            shouldShow = true;
+            animationCTS?.Cancel();
+            animationCTS?.Dispose();
+            animationCTS = new CancellationTokenSource();
+            FadeInOutlineAsync(animationCTS.Token).Forget();
+        }
+        else //We have to fade out and then clear the scriptable object
+        {
+            //We are already fade out (or animating towards it)
+            if (!shouldShow)
+                return;
+
+            shouldShow = false;
+            animationCTS?.Cancel();
+            animationCTS?.Dispose();
+            animationCTS = new CancellationTokenSource();
+            FadeOutOutlineAsync(animationCTS.Token).Forget();
+        }
+    }
+
+    private async UniTaskVoid FadeInOutlineAsync(CancellationToken ct = default)
+    {
+        const float SPEED = 1 / 0.333f; // 300ms
+        const float SHOW_FADE = 1;
+
+        var settings = outlineScreenEffectFeature.settings;
+
+        try
+        {
+            ct.ThrowIfCancellationRequested();
```

Comment:
let's move it outside of `try` and just return if `IsCancellationRequestion`, I don't see the reason to go through the exceptions handling mechanism

Reply:
Yes!

Reply:
I like using the exception mechanism because it prevenst you for having to do the tiresome boilerplate
```
if(ct.IsCancellationRequested) return;
```

After every `await`. But in this case, it's nicely wrapped within the while